l3build-aux.lua

```diff
@@ -57,69 +75,89 @@ function setepoch()
       .. os_concat
 end
 
-local function getscriptname()
+---Returns the script name depending on the calling sequence.
+---`l3build ...` -> full path of `l3build.lua` in the TDS
+---When called via `texlua l3build.lua ...`, `l3build.lua` is resolved to either 
+---`./l3build.lua` or the full path of `l3build.lua` in the TDS.
+---`texlua l3build.lua` -> `/Library/TeX/texbin/l3build.lua` or `./l3build.lua`
+---@return string
+local function get_script_name()
   if match(arg[0], "l3build$") or match(arg[0], "l3build%.lua$") then
     return lookup("l3build.lua")
   else
-    return arg[0]
+    return arg[0] -- Why no lookup here? 
   end
 end
 
--- Do some subtarget for all modules in a bundle
-function call(dirs, target, opts)
-  -- Turn the option table into a string
-  local opts = opts or options
-  local s = ""
+-- Performs the task named target given modules in a bundle.
+---A module is the path of a directory relative to the main one.
+---Uses `run` to launch a command.
+---@param modules table List of modules.
+---@param target string
+---@param opts table
+---@return number 0 on proper termination, a non 0 error code otherwise.
+---@see many places, including latex2e/build.lua
+---@usage Public
+function call(modules, target, opts)
+  -- Turn the option table into a CLI option string
+  opts = opts or options
+  local cli_opts = ""
   for k,v in pairs(opts) do
-    if k ~= "names" and k ~= "target" then -- Special cases
-      local t = option_list[k] or { }
-      local arg = ""
+    if k ~= "names" and k ~= "target" then -- Special cases, to be removed!!!
```

Comment:
I'm not sure why you'd want to remove these

Reply:
Not a remove.
`arg` has been renamed to `value` because `arg` is already used by Lua, it may be misleading.

Reply:
You added `to be removed!!!`, which is what I didn't follow. 

On using `arg`, this is local so it shouldn't be a big issue (we don't need the CLI `arg`  table here), but I'm not opposed to a rename.

Reply:
Oups!!! `target` and `names` contradict the fact that `options` keys are CLI options. Replacing them by global variables or replacing the keys by "target\0" and "names\0" would resolve that contradiction.

The rename is a detail but details really make life easier. If we want to know every place where the global `arg` is used, then no need to manually filter out a find output.

Reply:
I guess this is a matter of how one things about it: I see `options` as 'everything after `l3build`' on the CLI. For example, there is no certainty that the 'names' really are names, they could be some other optional data (OK, then perhaps better called 'final list' or similar).

Reply:
That said, I have no issue with a change _provided_ it's documented: we do currently say that everything, including the target and the list of names, is in that table.